### PR TITLE
Test GPU reductions for extruded spaces

### PR DESCRIPTION
**Pull request by sriharshakandala**
- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Review by simonbyrne — APPROVED** [review]

File: test/Fields/reduction_cuda.jl [hunk 1]
```diff
@@ -115,3 +130,106 @@ end
...
+
+    @test Base.maximum(identity, Yc)[] ≈ Base.maximum(identity, Yc_cpu)
+    @test Base.minimum(identity, Yc)[] ≈ Base.minimum(identity, Yc_cpu)
+
+    @test Statistics.mean(Yc)[] ≈ Statistics.mean(Yc_cpu)
+
+    @test LinearAlgebra.norm(Yc, 1)[] ≈ LinearAlgebra.norm(Yc_cpu, 1)
+    @test LinearAlgebra.norm(Yc, 2)[] ≈ LinearAlgebra.norm(Yc_cpu, 2)
+    @test LinearAlgebra.norm(Yc, 3)[] ≈ LinearAlgebra.norm(Yc_cpu, 3)
+    @test LinearAlgebra.norm(Yc, Inf)[] ≈ LinearAlgebra.norm(Yc_cpu, Inf)
```
Comment (simonbyrne) [inline test/Fields/reduction_cuda.jl:221]: can these return a scalar?

**Review by simonbyrne — CHANGES_REQUESTED** [review]

File: test/Fields/reduction_cuda.jl (hunk 1 above)
Comment (sriharshakandala) [reply test/Fields/reduction_cuda.jl:221]: It can currently handle mutiple fields. Should we specialize this for just `Nf = 1`?

Comment (sriharshakandala) [reply test/Fields/reduction_cuda.jl]: Modified to return a scalar (result for the first field)

**Comment by sriharshakandala** [comment]
bors try

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**